### PR TITLE
Move to Scala 2.13

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR updates Strimzi to use Scala 2.13 instead of 2.12. 

As discussed on Slack and on the Strimzi Community Call, the OPA Authorizer was upgraded to a new version which supports Scala 2.13 and introduce new format of the input data. This was noted in the CHANGELOG.md file.

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -108,7 +108,7 @@
         <!-- keep in-sync with dataformat-yaml -->
         <json-path.version>4.1.1</json-path.version>
         <zkclient.version>0.11</zkclient.version>
-        <scala-library.version>2.12.12</scala-library.version>
+        <scala-library.version>2.13.5</scala-library.version>
```
Comment (tomncooper) [inline pom.xml:111]: Any reason not to use the latest `2.13.6`?

**Review by tomncooper — APPROVED** [review]
LGTM, other than maybe bumping to the latest scala point release.

**Review by tombentley — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (scholzj) [reply pom.xml:111]: It probably did not exist when I opened the PR for the first time.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,12 +2,17 @@
 
 ## 0.25.0
 
+* Move from Scala 2.12 to Scala 2.13
+* Update Open Policy Agent Authorizer to new version supporting Scala 2.13. See the _Changes, deprecations and removals_ sections for more details.
```
Comment (PaulRMellor) [inline CHANGELOG.md:6]:
```suggestion
* Open Policy Agent authorizer updated to a new version supporting Scala 2.13. See the _Changes, deprecations and removals_ sections for more details.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,12 +2,17 @@
...
 * Allow a custom password to be set for SCRAM-SHA-512 users by referencing a secret in the `KafkaUser` resource
 
 ### Changes, deprecations and removals
 
 * The `KafkaConnectS2I` resource has been removed and is no longer supported by the operator.
   Please use the [migration guide](https://strimzi.io/docs/operators/0.24.0/full/using.html#proc-migrating-kafka-connect-s2i-str) to migrate your `KafkaConnectS2I` deployments to [`KafkaConnect` Build](https://strimzi.io/docs/operators/latest/full/deploying.html#creating-new-image-using-kafka-connect-build-str) instead.
+* The Open Policy Agent Authorizer has been updated to a new version which supports Scala 2.13.
```
Comment (PaulRMellor) [inline CHANGELOG.md:13]:
```suggestion
* The Open Policy Agent authorizer has been updated to a new version that supports Scala 2.13.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,12 +2,17 @@
...
+  The new release introduces a new format of the input data sent to Open Policy Agent.
```
Comment (PaulRMellor) [inline CHANGELOG.md:14]:
```suggestion
  The new release introduces a new format of the input data sent to the Open Policy Agent server.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,12 +2,17 @@
...
+  For more information about the new format and about the migration from the old version, please see https://github.com/Bisnode/opa-kafka-plugin/releases/tag/v1.0.0
```
Comment (PaulRMellor) [inline CHANGELOG.md:15]:
```suggestion
  For more information about the new format and how to migrate from the old version, see the [OPA v1.0.0 release notes](https://github.com/Bisnode/opa-kafka-plugin/releases/tag/v1.0.0)
```

File: documentation/api/io.strimzi.api.kafka.model.KafkaAuthorizationOpa.adoc [hunk 6]
```diff
@@ -1,5 +1,7 @@
 To use link:https://www.openpolicyagent.org/[Open Policy Agent^] authorization, set the `type` property in the `authorization` section to the value `opa`,
 and configure OPA properties as required.
+Strimzi uses the Kafka authorization plugin from Bisnode as the Open Policy Agent authroizer.
```
Comment (PaulRMellor) [inline documentation/api/io.strimzi.api.kafka.model.KafkaAuthorizationOpa.adoc:3]:
```suggestion
Strimzi uses Bisnode's Kafka authorization plugin as the Open Policy Agent authorizer.
```

**Review by PaulRMellor — APPROVED** [review]
Looks good. Couple of suggestions. :+1:

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,12 +2,17 @@
 
 ## 0.25.0
 
+* Move from Scala 2.12 to Scala 2.13
```
Comment (scholzj) [inline CHANGELOG.md:5]:
```suggestion
* Move from Scala 2.12 to Scala 2.13. (#5192)
```

**Pull request merged**